src/services/event-producer/sqs/sqs-event-producer-service.ts

```diff
@@ -0,0 +1,34 @@
+import 'reflect-metadata'
+import { Config } from 'node-config-ts'
+import { inject, singleton } from 'tsyringe'
+import { SQSClient, SendMessageCommand } from '@aws-sdk/client-sqs'
+import { EventProducerService } from '../event-producer-service.js'
+import { v4 as uuidv4 } from 'uuid'
+
+@singleton()
+export class SQSEventProducerService implements EventProducerService {
+  private readonly sqsClient: SQSClient
+  constructor(@inject('config') private config?: Config) {
+    this.sqsClient = new SQSClient({})
+  }
+
+  /**
+   * Emits an anchor event by sending a message to the configured SQS
+   */
+  public async emitAnchorEvent(): Promise<void> {
```

Comment:
Would it make sense for this method to accept the body as an argument? That way the caller can decide what to send in the event without needing to worry about the underlying queue technology.

Then, you'll also be able to read the event off the queue in your unit tests and make sure the body was correct.